### PR TITLE
[Motions 2022 11 lwg 12] P1264R2 Revising the wording of stream input operations

### LINKED ISSUE #5973: [2022-11 LWG Motion 12] P1264R2 Revising the wording of stream input operations
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p1264r2.pdf

### PR DESCRIPTION
**Pull request by burblebee**
Fixes #5973.
Fixes https://github.com/cplusplus/papers/issues/342.

### CODE REVIEW

**Review by JohelEGP — APPROVED** [review]

File: source/iostreams.tex [hunk 1]
```diff
@@ -5044,35 +5007,28 @@
...
-If
-\tcode{(exceptions()\&badbit) != 0}
-then the exception is rethrown.
-It also counts the number of characters extracted.
-If no exception has been thrown it ends
-by storing the count in a member object
-and returning the value specified.
+\tcode{ios_base::badbit} is set in the local error state,
+\tcode{*this}'s error state is set to the local error state, and
+the exception is rethrown if \tcode{(exceptions() \& badbit) != 0}.
```
Comment (jwakely) [inline source/iostreams.tex:5025]: Another pre-existing use of `badbit` instead of `ios_base::badbit`.

**Review by jwakely — CHANGES_REQUESTED** [review]
Three changes requested, and two general comments.

File: source/iostreams.tex [hunk 2]
```diff
@@ -4649,20 +4626,22 @@
...
-\end{footnote}
-in
-\tcode{*this}'s
-error state.
-If
-\tcode{(exceptions()\&badbit) != 0}
-then the exception is rethrown.
+\tcode{ios::badbit} is set in the local error state,
+\tcode{*this}'s error state is set to the local error state, and
+the exception is rethrown if \tcode{(exceptions() \& badbit) != 0}.
```
[the remark this answers is not in the document]
Comment (tkoeppe) [reply source/iostreams.tex:4653]: Would you like to have an issue for this?

File: source/iostreams.tex (hunk 1 above)
Comment (tkoeppe) [reply source/iostreams.tex:5037]: #6019

File: source/iostreams.tex (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkoeppe) [reply source/iostreams.tex:4653]: #6019

**Pull request merged**